### PR TITLE
FastEmbed integration with FlagEmbedding

### PR DESCRIPTION
**Pull request by NirantK**

### CODE REVIEW

File: extractors/indexify_extractors/embedding_extractor.py [hunk 1]
```diff
@@ -45,6 +45,31 @@ def _create_splitter(self, text_splitter: str) -> Callable[[str], List[str]]:
     def extract_embeddings(self, texts: List[str]) -> List[List[float]]:
         ...
 
+
+class FlagEmbedding(BaseEmbeddingExtractor):
+    def __init__(self):
+        self.embedding_model = FastFlagEmbedding(model_name="BAAI/bge-small-en", max_length=512)
```
Comment (diptanu) [inline extractors/indexify_extractors/embedding_extractor.py:51]: The import to `FastFlagEmbedding` is missing.

**Review by diptanu — CHANGES_REQUESTED** [review]
The design looks great! There is an import missing.

File: extractors/indexify_extractors/embedding_extractor.py (hunk 1 above)
Comment (NirantK) [reply extractors/indexify_extractors/embedding_extractor.py:54]: Just added a new import statement which should resolve this

File: extractors/indexify_extractors/embedding_extractor.py [hunk 2]
```diff
@@ -3,6 +3,9 @@
 from typing import Any, Optional, List, Literal, Callable, Union
 
 from dataclasses import dataclass
+from typing import Any, Callable, List, Literal, Optional, Union
+
+from fastembed import FlagEmbedding as FastFlagEmbedding
```
Comment (diptanu) [inline extractors/indexify_extractors/embedding_extractor.py:8]:
```
from fastembed.embedding import FlagEmbedding as FastFlagEmbedding
```
@NirantK I think the import needs to be updated to that, or add an __init__.py in the package which rexports the symbols appropriately.

**Review by diptanu — CHANGES_REQUESTED** [review]

File: extractors/indexify_extractors/embedding_extractor.py [hunk 3]
```diff
@@ -45,6 +48,31 @@ def _create_splitter(self, text_splitter: str) -> Callable[[str], List[str]]:
     def extract_embeddings(self, texts: List[str]) -> List[List[float]]:
         ...
 
+
+class FlagEmbedding(BaseEmbeddingExtractor):
+    def __init__(self):
```
Comment (diptanu) [inline extractors/indexify_extractors/embedding_extractor.py:53]: Have to initialize the super class here - 
```
 super(FlagEmbedding, self).__init__(max_context_length=384)
```

**Review by diptanu — CHANGES_REQUESTED** [review]

**Pull request merged**